### ISSUE TITLE
Running migration with umzug has some issues

### ISSUE THREAD

**Issue by HcgRandon** [cockroachdb/sequelize-cockroachdb#19]
When running the following migration

```js
'use strict';

module.exports = {
 up: (query, types) => {
  query.addColumn('Users', 'lastName', types.STRING);
 },
 down: (query, types) => {
  query.removeColumn('Users', 'lastName');
 },
};
```
This error is thrown `SequelizeDatabaseError: database "public" does not exist`
the query that it attempt to ran was this:
```
Executing (default): ALTER TABLE "public"."Users" ADD COLUMN "lastName" VARCHAR(255);
```

After digging into the code a little bit I think i tracked this down to this line here in sequelize
https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query-generator.js#L26

You can see they call this function in the postgres query-generator dialect on this line
https://github.com/sequelize/sequelize/blob/master/lib/dialects/postgres/query-generator.js#L241

Versions:
Cockroachdb: CCL v1.1.0-rc.1 @ 2017/10/05 15:33:25 (go1.8.3)
sequelize-cockroachdb: 1.0.2
umzug: 2.0.1

**Comment by benesch** [comment]
As a workaround, you might have some luck with this:

```
query.addColumn({tableName: 'Users', schema: 'YOURDBNAME'}, 'lastName', types.STRING);
```

More permanently, we're still sorting out some impedance mismatches between our notion of namespaces and Postgres's. See https://github.com/cockroachdb/cockroach/pull/13319 for an overview of the situation. 

/cc @jordanlewis @knz is there anything planned to fix this?

**Comment by HcgRandon** [comment]
Sadly this isn't really a viable option for my current setup. The name of my database changes based on the instance my application is running as at the time.

Thanks for the workaround though!

**Comment by knz** [comment]
Yes we probably want to do something about this. I talked with Jordan and there is a way to introduce the public namespace even if we don't do all the work described in #13319.

**Comment by benesch** [comment]
Hmm, perhaps we can attack this from the Sequelize side as well. Looks like we might have some luck by setting `_dialect.supports.schemas` to `false`. I'll look into this more.

**Comment by HcgRandon** [comment]
Any updates on this. I really want to start developing and using cockroach in my application. But this issue is preventing me from doing so

**Comment by benesch** [comment]
Sorry, not yet. It's worth noting, though, that the workaround I suggested before could support dynamic databases:

```
query.addColumn({tableName: 'Users', schema: connArgs.database}, 'lastName', types.STRING);
```

Specifically, in order to connect to a given database, some component of your application must know the database's name. It should be possible to plumb that database name into the migration, but perhaps your setup makes this prohibitive.
